From 3bbfbed005b5a45c23763d08225b7e239559d29d Mon Sep 17 00:00:00 2001
From: Koopa <codingkoopa@gmail.com>
Date: Tue, 7 Sep 2021 02:44:19 -0400
Subject: [PATCH] patch for comet-observatory

- rewrite .gitignore
- remove license, readme, desktop file, manpage
- remove terminfo patch
- remove case for handling nonexistent config
- add symlink to comet-observatory config
- add patching support
- add desired patches
- disable ccache to prevent unwanted caching of config.h
---
 .SRCINFO            |  8 ++--
 .gitignore          |  7 +---
 PKGBUILD            | 62 +++++++------------------------
 README.terminfo.rst | 90 ---------------------------------------------
 config.h            |  1 +
 terminfo.patch      | 10 -----
 6 files changed, 21 insertions(+), 157 deletions(-)
 delete mode 100644 README.terminfo.rst
 create mode 120000 config.h
 delete mode 100644 terminfo.patch

diff --git a/.SRCINFO b/.SRCINFO
index 801c06a..402e927 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -10,10 +10,10 @@ pkgbase = st
 	license = MIT
 	depends = libxft
 	source = https://dl.suckless.org/st/st-0.8.4.tar.gz
-	source = terminfo.patch
-	source = README.terminfo.rst
+	source = config.h
+	source = https://st.suckless.org/patches/alpha/st-alpha-0.8.2.diff
 	sha256sums = d42d3ceceb4d6a65e32e90a5336e3d446db612c3fbd9ebc1780bc6c9a03346a6
-	sha256sums = f9deea445a5c6203a0e8e699f3c3b55e27275f17fb408562c4dd5d649edeea23
-	sha256sums = 0ebcbba881832adf9c98ce9fe7667c851d3cc3345077cb8ebe32702698665be2
+	sha256sums = SKIP
+	sha256sums = SKIP
 
 pkgname = st
diff --git a/.gitignore b/.gitignore
index 2d32340..f2eb025 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,3 @@
-/*.pkg.tar.xz
-/config.def.h
-/config.h
-/pkg/
-/src/
 /st-*.tar.gz
+/*.diff
+/*.pkg.*
diff --git a/PKGBUILD b/PKGBUILD
index 95127d8..ac16242 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -12,61 +12,27 @@ license=('MIT')
 depends=(libxft)
 url=https://st.suckless.org
 source=(https://dl.suckless.org/$pkgname/$pkgname-$pkgver.tar.gz
-        terminfo.patch
-        README.terminfo.rst)
+        config.h
+        https://st.suckless.org/patches/alpha/st-alpha-0.8.2.diff
+)
 sha256sums=('d42d3ceceb4d6a65e32e90a5336e3d446db612c3fbd9ebc1780bc6c9a03346a6'
-            'f9deea445a5c6203a0e8e699f3c3b55e27275f17fb408562c4dd5d649edeea23'
-            '0ebcbba881832adf9c98ce9fe7667c851d3cc3345077cb8ebe32702698665be2')
-_sourcedir=$pkgname-$pkgver
-_makeopts="--directory=$_sourcedir"
+            'SKIP'
+            'SKIP'
+)
 
 prepare() {
-  patch --directory="$_sourcedir" --strip=0 < terminfo.patch
-
-  # This package provides a mechanism to provide a custom config.h. Multiple
-  # configuration states are determined by the presence of two files in
-  # $BUILDDIR:
-  #
-  # config.h  config.def.h  state
-  # ========  ============  =====
-  # absent    absent        Initial state. The user receives a message on how
-  #                         to configure this package.
-  # absent    present       The user was previously made aware of the
-  #                         configuration options and has not made any
-  #                         configuration changes. The package is built using
-  #                         default values.
-  # present                 The user has supplied his or her configuration. The
-  #                         file will be copied to $srcdir and used during
-  #                         build.
-  #
-  # After this test, config.def.h is copied from $srcdir to $BUILDDIR to
-  # provide an up to date template for the user.
-  if [ -e "$BUILDDIR/config.h" ]
-  then
-    cp "$BUILDDIR/config.h" "$_sourcedir"
-  elif [ ! -e "$BUILDDIR/config.def.h" ]
-  then
-    msg='This package can be configured in config.h. Copy the config.def.h '
-    msg+='that was just placed into the package directory to config.h and '
-    msg+='modify it to change the configuration. Or just leave it alone to '
-    msg+='continue to use default values.'
-    echo "$msg"
-  fi
-  cp "$_sourcedir/config.def.h" "$BUILDDIR"
+  cd "$srcdir/$pkgname-$pkgver"
+  # Beware that this is vulnerable to applying pre-existing patches - always use makepkg -C here.
+  git apply -v "$srcdir"/*.diff
+  ln -sf "$srcdir/config.h" config.h
 }
 
 build() {
-  make $_makeopts X11INC=/usr/include/X11 X11LIB=/usr/lib/X11
+  cd "$srcdir/$pkgname-$pkgver"
+  CCACHE_DISABLE=1 make X11INC=/usr/include/X11 X11LIB=/usr/lib/X11
 }
 
 package() {
-  local installopts='--mode 0644 -D --target-directory'
-  local shrdir="$pkgdir/usr/share"
-  local licdir="$shrdir/licenses/$pkgname"
-  local docdir="$shrdir/doc/$pkgname"
-  make $_makeopts PREFIX=/usr DESTDIR="$pkgdir" install
-  install $installopts "$licdir" "$_sourcedir/LICENSE"
-  install $installopts "$docdir" "$_sourcedir/README"
-  install $installopts "$docdir" README.terminfo.rst
-  install $installopts "$shrdir/$pkgname" "$_sourcedir/st.info"
+  cd "$srcdir/$pkgname-$pkgver"
+  make PREFIX=/usr DESTDIR="$pkgdir" install
 }
diff --git a/README.terminfo.rst b/README.terminfo.rst
deleted file mode 100644
index cd724d0..0000000
--- a/README.terminfo.rst
+++ /dev/null
@@ -1,90 +0,0 @@
-Terminfo Entries
-================
-
-:Author: Sebastian J. Bronner <waschtl@sbronner.com>
-:Date: 2019-11-06
-:URL: https://aur.archlinux.org/packages/st
-
-The Problem
------------
-
-Most Linux-based distributions will have their terminfo entries in the
-``ncurses`` package. These live in ``/usr/share/terminfo``. There you will find
-several entries for ``st``. As of ``ncurses-6.1`` as packaged in ArchLinux,
-these, do not work well with current versions of ``st``.
-
-Specifically, I have observed the following issues when using the terminfo
-entries supplied with ``ncurses``:
-
-* Start ``st``.
-* Run ``tmux`` in the ``st`` window.
-* Run ``nvim`` in ``tmux``.
-
-  * ``tmux`` will crash immediately with the following message:
-
-    .. code:: console
-
-       [lost server]
-                    %
-
-* Run ``w3m`` with any URL in ``tmux``.
-
-  * ``w3m`` will not react to any key presses (most notably the arrow keys and
-    ``q``) and must be terminated with ``killall w3m``.
-  * The command line returned where ``w3m`` was running will show all the
-    missing keypresses.
-
-For reference, I performed these tests using the following software versions:
-``st`` 0.8.2, ``tmux`` 2.9_a, ``neovim`` 0.4.2, and ``w3m``
-0.5.3.git20190105-1.
-
-A couple of issues have been filed against ``tmux`` for this. Although it is
-not a problem with ``tmux`` the issues remain for reference:
-
-| https://github.com/tmux/tmux/issues/1264
-| https://github.com/tmux/tmux/issues/1593
-
-Complications
--------------
-
-While it may be tempting to use the terminfo files shipped with ``st``, that
-would be a bad idea from a packaging standpoint. There are two reasons for
-this: (1) As of ``st-0.8.2`` they have errors in them. This has been fixed and
-will roll out in the next release, though.
-
-| https://lists.gnu.org/archive/html/bug-ncurses/2019-10/msg00071.html
-| https://lists.suckless.org/hackers/1911/17006.html
-
-(2) It is vital that remote terminals connecting via ``ssh`` are known. That is
-accomplished by having all terminfo files in one package (like ``ncurses``).
-
-``ncurses`` has released updated terminfo entries. The problem as described
-above is fixed. However, the updated terminfo entries are not installed on
-ArchLinux because they were released as a standalone file without a new release
-of the rest of ``ncurses``.
-
-The Solution
-------------
-
-The solution must be to get the updated terminfo entries packaged for
-ArchLinux. I believe that the ``ncurses`` package should be updated to that
-effect.
-
-https://bugs.archlinux.org/task/57596
-
-The Workaround
---------------
-
-A somewhat painful, albeit pragmatic, workaround is as follows: A user that
-encounters issues that are due to mismatched terminfo entries, such as those
-above, can populate a user-level terminfo database from the terminfo entries
-supplied by ``st``. The following command will do this:
-
-.. code:: shell
-
-   tic -sx /usr/share/st/st.info
-
-The generated terminfo database will most likely be placed in ``~/.terminfo``.
-The command will print the actual location used. Unfortunately, these files
-need to be kept up to date by hand. Actually they should probably be removed as
-soon as a new version of ``ncurses`` is released.
diff --git a/config.h b/config.h
new file mode 120000
index 0000000..499a1f5
--- /dev/null
+++ b/config.h
@@ -0,0 +1 @@
+/home/koopa/Documents/Projects/Bash/comet-observatory/config/st.h
\ No newline at end of file
diff --git a/terminfo.patch b/terminfo.patch
deleted file mode 100644
index 71dc270..0000000
--- a/terminfo.patch
+++ /dev/null
@@ -1,10 +0,0 @@
---- Makefile.old	2019-02-09 12:50:41.000000000 +0100
-+++ Makefile	2019-11-07 00:00:24.487953923 +0100
-@@ -47,7 +47,6 @@
- 	mkdir -p $(DESTDIR)$(MANPREFIX)/man1
- 	sed "s/VERSION/$(VERSION)/g" < st.1 > $(DESTDIR)$(MANPREFIX)/man1/st.1
- 	chmod 644 $(DESTDIR)$(MANPREFIX)/man1/st.1
--	tic -sx st.info
- 	@echo Please see the README file regarding the terminfo entry of st.
- 
- uninstall:
-- 
2.33.1

